felix/bpf-gpl/types.h

```diff
@@ -120,22 +120,27 @@ struct cali_tc_ctx {
   struct fwd fwd;
 };
 
-static CALI_BPF_INLINE struct ethhdr* tc_ethhdr(struct cali_tc_ctx *ctx)
+static CALI_BPF_INLINE struct iphdr* ipv4hdr(struct cali_tc_ctx *ctx)
```

Comment:
It may come in some next iteration, but I wonder whether it would be better to call it just `iphdr` without version and compile the function so that it returns the right type depending whether you use it in the v4 or v6 version of the program.

I'd assume that lots of the code with the substitutions could be reused for v6. If it turns out not, we could just do the v4 subst then or just add v6 where necessary :thinking:

Reply:
Yes, there are more PRs coming in the area. Initially, I tried to use a common function/macro like `iphdr` for both v4 and v6. However:
- To make it work in a function, I had to use `if` condition checking the version. But this approach would impact the performance since each access executes an `if` condition. This is mainly due to the fact that we know the packet version during run time, not compile time.
- I could not use macros, since again macros can only help with compile time changes.

For that, I was thinking to start with this approach, and make adjustment as we go forward and make decision based on one particular part of code. Does it make sense?